atom/browser/api/atom_api_window.cc

```diff
@@ -584,8 +584,17 @@ void Window::SetFocusable(bool focusable) {
   return window_->SetFocusable(focusable);
 }
 
-void Window::SetProgressBar(double progress) {
-  window_->SetProgressBar(progress);
+void Window::SetProgressBar(double progress, mate::Arguments* args) {
+  std::string mode;
+
+  if (args && args->Length() == 2) {
```

Comment:
We are currently strictly following Chromium's coding styles, for introducing things like `gsl::not_null`, you need to convince Chromium team to adopt it first.